### ISSUE TITLE
Lean unstable

### ISSUE THREAD

**Issue by htzh** [leanprover/lean3#708]
I am having a stability problem similar to #593 when editing the following file at the line indicated:

https://github.com/htzh/leanproved/blob/9cead936b369aa1c96c83224e37325bd87613002/group_theory/pgroup.lean#L52

I have multiple Windows VS debugger popup messages saying Lean crashed. I will remove the offending rewrite command (I am not sure that the syntax is legal) but thought I should report this as a stability bug.

**Comment by leodemoura** [comment]
Dear Haitao,

I'm still in the "black hole" and overwhelmed by the number of issues. 
Any additional information would be very welcome. 
Example: what is the first commit that produces the instability?
A small sequence of operations that produces the VS debugger popup.

Thanks,
Leo

**Comment by htzh** [comment]
This is not an urgent issue and can be dealt with later. The crash only happens at the spot I referred to in the opening post. 
https://github.com/htzh/leanproved/blob/9cead936b369aa1c96c83224e37325bd87613002/group_theory/pgroup.lean#L52

```
 begin rewrite {a::l}Prodl at Pprod end
```

It is very repeatable. Simply opening the file in Emacs would bring on the crash message. As I said I am not sure this is legal syntax. After I removed the `{}` pattern spec the crash no longer happens. Current version works with no problem:

```
    rewrite (Prodl_cons id a l) at Pprod,
```
